### ISSUE TITLE
If esbuild is in watch mode and there is an error in sass, the plugin stops working

### ISSUE THREAD

**Issue by ArthurHeitmann** [glromeo/esbuild-sass-plugin#20]
So far everything has been working great except for this one thing.

My setup:

```JS
import esbuild from "esbuild";
import { sassPlugin } from "esbuild-sass-plugin";

esbuild.build({
	entryPoints: [
		"test.scss",
	],
	bundle: false,
	outfile: "test.css",
	sourcemap: true,
	plugins: [
		sassPlugin({
		})
	],
	watch: (error, result) => {
		if (error)
			console.error("rebuild SASS failed")
		else
			console.log("rebuild SASS succeeded")
	}
}).catch(() => process.exit(1));
```

If I do any valid changes to test.scss, everything works great.

When there is an error in tests.scss (missing semicolon), the error is displayed as expected. But when fixing the error or doing any other changes to test.scss nothing happens. You then have to manually restart the script.

**Comment by glromeo** [comment]
I could fix this issue by handling more gracefully the errors but at best I would have to cache the watchedFiles from the last successful build which I'd rather not do because esbuild does it already.
It wouldn't make the plugin slower but it would make it more memory hungry and I don't like that.
Let's keep this issue open until we see where esbuild goes with the issue #1063, I will then provide a fix accordingly.

**Comment by ArthurHeitmann** [comment]
I see the problem. In the meanwhile I have a fix on my [fork](https://github.com/ArthurHeitmann/esbuild-sass-plugin). I don't really know much about sass compilation or esbuild plugins, so I don't know if I missed something.

> Unused memory is wasted memory

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-- someone on reddit

**Comment by glromeo** [comment]
Given the cache is used only in watch mode I ended up providing a fix in v1.4.10

**Comment by ArthurHeitmann** [comment]
Seems we were thinking the same things. Thank you

**Comment by glromeo** [comment]
Sorry I wasn't paying attention... your PR would have been welcome.

**Comment by ArthurHeitmann** [comment]
Did you forget to run `tsc` before publishing? Because in the plugin.ts [it says](https://github.com/glromeo/esbuild-sass-plugin/blob/main/src/plugin.ts#L197) `build.initialOptions` and in plugin.js [it just says](https://github.com/glromeo/esbuild-sass-plugin/blob/main/lib/plugin.js#L174) `build`. And btw. the watch parameter can also be false :)

**Comment by glromeo** [comment]
@ArthurHeitmann thanks for spotting it! I need to improve my publish pipeline 🤦‍♂️

**Comment by glromeo** [comment]
Ok, fixed that in v1.4.11
